tools/qr_coder/qr_coder.xml

```diff
@@ -0,0 +1,65 @@
+<tool id="qr_coder_1" name="qr_coder" version="1.0">
+   <description>Generate qr code images</description>
+   <requirements>
+      <requirement type="package" version="5.1">qrcode</requirement>
+      <requirement type="package" version="1.4.1">image</requirement>
+   </requirements>
+   <stdio>
+      <exit_code range="1:" />
+      <regex match="Error:" />
+      <regex match="Exception:" />
+   </stdio>
+   <command interpreter="python">
+      qr_coder.py -qrData "$qrData" -qrOutput $qrOutput
+      #if $qrVersion == ""
+         --qrFit
+      #else
+         -qrVersion $qrVersion
+      #end if
+       #if $qrBorder
+         -qrBorder $qrBorder
+      #end if
+      #if $qrBoxSize
+         -qrBoxSize $qrBoxSize
+      #end if
+      #if $qrErrorCorrection
+         -qrErrorCorrection $qrErrorCorrection
+      #end if
+   </command>
+   <inputs>
+      <param name="qrData"            type="text"    label="Data to be coded as an qr image"                                           optional="false" value="https://mississippi.snv.jussieu.fr/artbio/"/>
```

Comment:
The value corresponds to what will be pre-selected. It's probably better to not have a value, which will require the user to enter something. Also note that there is a help="" tag. You could do for example help="Enter the text for which you whish to generate a QR code."